modules/appeal/src/main/AppealApi.scala

```diff
@@ -53,36 +53,48 @@ final class AppealApi(
 
   def countUnread = coll.countSel($doc("status" -> Appeal.Status.Unread.key))
 
-  def myQueue(using me: Me) = bothQueues(snoozer snoozedKeysOf me.userId map (_.appealId.userId))
+  def myQueue(filter: Option[Appeal.Filter])(using me: Me) =
+    bothQueues(filter, snoozer snoozedKeysOf me.userId map (_.appealId.userId))
 
-  private def bothQueues(exceptIds: Iterable[UserId]): Fu[List[Appeal.WithUser]] =
+  private def bothQueues(
+      filter: Option[Appeal.Filter],
+      exceptIds: Iterable[UserId]
+  ): Fu[List[Appeal.WithUser]] =
     fetchQueue(
       selector = $doc("status" -> Appeal.Status.Unread.key) ++ {
         exceptIds.nonEmpty so $doc("_id" $nin exceptIds)
       },
+      filter = filter,
       ascending = true,
       nb = 50
     ) flatMap { unreads =>
       fetchQueue(
         selector = $doc("status" $ne Appeal.Status.Unread.key),
+        filter = filter,
         ascending = false,
         nb = 60 - unreads.size
       ) map { unreads ::: _ }
     }
 
-  private def fetchQueue(selector: Bdoc, ascending: Boolean, nb: Int): Fu[List[Appeal.WithUser]] =
+  private def fetchQueue(
+      selector: Bdoc,
+      filter: Option[Appeal.Filter],
+      ascending: Boolean,
+      nb: Int
+  ): Fu[List[Appeal.WithUser]] =
     coll
       .aggregateList(maxDocs = nb, _.sec): framework =>
         import framework.*
         Match(selector) -> List(
           Sort((if ascending then Ascending.apply else Descending.apply) ("firstUnrepliedAt")),
           Limit(nb),
```

Comment:
I think keeping `Limit(nb)` might cause an issue where someone wants to filter appeals that are not in the first `nb` (let's say for example clean marks, which are seldom), but I wanted to wait to see how it goes in practice before removing or making the number higher